src/main/scala/com/twitter/finagle/websocket/Frame.scala

```diff
@@ -0,0 +1,19 @@
+package com.twitter.finagle.websocket
+
+import com.twitter.io.Buf
+
+/**
+ * Represents various WebSocket frames.
+ *
+ * This is a simplification of the frame types described in RFC6455[1]. Notably
+ * absent are Continuation and Close. Close is handled directly in the
+ * pipeline, initiating the close handshake. Continuations are treated as
+ * Binary frames, which means we lose the ability to determine fragmentation.
+ *
+ * [1]: https://tools.ietf.org/html/rfc6455
+ */
+sealed trait Frame
+case class Text(text: String) extends Frame
```

Comment:
So glad you stepped in, @vkostyukov! I think the API needs some revision and work. I wonder if @sprsquish wouldn't mind having a "next" branch where we can stage our incremental PRs until the set of them are ready.